lib/queryLLM.ts

```diff
@@ -78,10 +79,31 @@ export async function streamLLMResponse(
   return response.body;
 }
 
+export function removeIdsFromMessages(messages: ChatGPTMessage[]): {
+  cleanedMessages: ChatGPTMessage[];
+  idStore: IDStore;
+} {
```

Comment:
mmm I can do thi,s but removeIDs is already a pretty complicated function. If we add an extra for loop and if statement in there and the logic around persisting idStore between different messages i think it will be much harder to follow. I quite like the design of writing general purpose testable utility functions then writing extra code to apply them to our specific use case. Basically i can do this refactor but i don't think it will improve the code very much

Reply:
Won't the idStore persistence logic become simpler? At the moment we have to pass it in every call

And the test coverage will increase - it's possible this (or minor adjustments to this made in the future) introduces a bug specific to our use-case

Saying that, am happy to let this slide because it's pretty simple code anyway